internal/components/componentlist.go

```diff
@@ -0,0 +1,132 @@
+package components
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/kyma-incubator/reconciler/pkg/keb"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v3"
+)
+
+const defaultNamespace = "kyma-system"
+
+// ComponentList collects component definitions
+type ComponentList struct {
+	Prerequisites []keb.Component
+	Components    []keb.Component
+}
+
+// ComponentDefinition defines a component in components list
+type ComponentDefinition struct {
+	Name      string
+	Namespace string
+}
+
+// ComponentListData is the raw component list
+type ComponentListData struct {
```

Comment:
```suggestion
type componentListData struct {
```
It does not have to be an exported type